### PR TITLE
Updating definitions for combined-stream v1.0.7

### PR DESCRIPTION
**Pull request by konpikwastaken**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: **https://www.npmjs.com/package/combined-stream**

- [x] Increase the version number in the header if appropriate.

### CODE REVIEW

**Comment by konpikwastaken** [comment]
- Also moved from using namespace as stricter tslint rules will prevent returning namespace (CombinedStream as a non-rereachable type)
- Added support for `CombinedStream.append((next) => { next(...) })` method.

File: types/combined-stream/index.d.ts [hunk 1]
```diff
@@ -1,19 +1,29 @@
...
 // Project: https://github.com/felixge/node-combined-stream
-// Definitions by: Felix Geisendörfer <https://github.com/felixge>, Tomek Łaziuk <https://github.com/tlaziuk>
+// Definitions by: Felix Geisendörfer <https://github.com/felixge>, Tomek Łaziuk <https://github.com/tlaziuk>, Kon Pik <https://github.com/konpikwastaken>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
 /// <reference types="node" />
 
-import { Stream } from "stream";
+import { Stream } from 'stream';
+import { ReadStream } from 'fs';
```
Comment (tlaziuk) [inline types/combined-stream/index.d.ts]: where is this import used?

Comment (konpikwastaken) [reply types/combined-stream/index.d.ts]: Oops, not sure how that got added. Removing.

**Comment by konpikwastaken** [comment]
PR feedback addressed, removed unnecessary import. Do you see any other issues?

File: types/combined-stream/index.d.ts [hunk 2]
```diff
@@ -1,19 +1,28 @@
...
 
 /// <reference types="node" />
 
-import { Stream } from "stream";
+import { Stream } from 'stream';
 
-declare class CombinedStream extends Stream implements CombinedStream.Options {
+type Appendable = NodeJS.ReadableStream | NodeJS.WritableStream | Buffer | string | NextFunction;
+// tslint:disable-next-line:no-any-union
+type NextFunction = (next: (stream: Appendable) => any | undefined) => any;
```
Comment (PranavSenthilnathan) [inline types/combined-stream/index.d.ts]:
```suggestion
type NextFunction = (next: (stream: Appendable) => any) => any;
```
`any` should include `undefined`. Is there a reason to include it explicitly?

File: types/combined-stream/index.d.ts [hunk 3]
```diff
@@ -34,23 +43,18 @@ declare class CombinedStream extends Stream implements CombinedStream.Options {
...
-    function isStreamLike(stream: any): stream is Stream;
+    on(event: 'close' | 'end' | 'resume' | 'pause', cb: () => void): this;
+    on(event: 'error', cb: (err: Error) => void): this;
+    on(event: 'data', cb: (data: any) => void): this;
+    once(event: 'close' | 'end' | 'resume' | 'pause', cb: () => void): this;
+    once(event: 'error', cb: (err: Error) => void): this;
+    once(event: 'data', cb: (data: any) => void): this;
+
+    create(options?: Options): CombinedStream;
+    isStreamLike(stream: any): stream is Stream;
```
Comment (PranavSenthilnathan) [inline types/combined-stream/index.d.ts]: These two members are not defined on the prototype so they should only be static as below.

**Review by PranavSenthilnathan — CHANGES_REQUESTED** [review]

File: types/combined-stream/index.d.ts (hunk 3 above)
Comment (konpikwastaken) [reply types/combined-stream/index.d.ts]: Done.

File: types/combined-stream/index.d.ts (hunk 2 above)
Comment (konpikwastaken) [reply types/combined-stream/index.d.ts]: Resolved

**Comment by konpikwastaken** [comment]
ping @PranavSenthilnathan can you please review again at your convenience?

**Review by PranavSenthilnathan — APPROVED** [review]

**Pull request merged**